core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1505,7 +1505,7 @@ class UnifiedLog(@volatile var logStartOffset: Long,
   /**
    * Flush all local log segments
    */
-  def flush(): Unit = flush(logEndOffset)
+  def flush(): Unit = flush(logEndOffset + 1)
```

Comment:
Could we add a comment why we need to flush to logEndOffset + 1? Also, could we have a test that verifies the index file is present after an empty segment is rolled and the broker is shut down?

Reply:
Thanks for the comment. Will write it later. Was too busy yesterday and didn't have time to do so.